DelvUI/Interface/AstrologianHudWindow.cs

```diff
@@ -368,59 +365,271 @@ internal class AstrologianHudWindow : HudWindow {
+        private void DrawStar()
+        {
+            var starPreCookingBuff =
+                PluginInterface.ClientState.LocalPlayer.StatusEffects.Where(o => o.EffectId == 1224);
+            var starPostCookingBuff =
+                PluginInterface.ClientState.LocalPlayer.StatusEffects.Where(o => o.EffectId == 1248);
             var starDuration = 0f;
             const float starMaxDuration = 10f;
 
-            var xPos = CenterX - XOffset + BaseXOffset + StarBarX;
-            var yPos = CenterY + YOffset + BaseYOffset + StarBarY;
+            var xPos = CenterX - XOffset + _config.ASTBaseOffset.X + _config.ASTStarBarPosition.X;
+            var yPos = CenterY + YOffset + _config.ASTBaseOffset.Y + _config.ASTStarBarPosition.Y;
             var starColorSelector = EmptyColor;
 
-            if (starPreCookingBuff.Any()) {
+            if (starPreCookingBuff.Any())
+            {
                 starDuration = starMaxDuration - Math.Abs(starPreCookingBuff.First().Duration);
-                starColorSelector = StarEarthlyColor;
+                starColorSelector = _config.ASTStarEarthlyColor.Map;
             }
 
             if (starPostCookingBuff.Any()) {
                 starDuration = Math.Abs(starPostCookingBuff.First().Duration);
-                starColorSelector = StarGiantColor;
+                starColorSelector = _config.ASTStarGiantColor.Map;
             }
 
-            var builder = BarBuilder.Create(xPos, yPos, StarHeight, StarWidth);
+            var builder = BarBuilder.Create(xPos, yPos, _config.ASTStarBarSize.X, _config.ASTStarBarSize.Y);
 
             var bar = builder.AddInnerBar(starDuration, starMaxDuration, EmptyColor, starColorSelector)
-                             .SetTextMode(BarTextMode.Single)
-                             .SetBackgroundColor(EmptyColor["background"])
-                             .SetText(BarTextPosition.CenterMiddle, BarTextType.Current);
-
-            if (starColorSelector == StarGiantColor && ShowStarGlowBar) {
-                bar.SetGlowColor(StarGlowColor["base"]);
+                .SetTextMode(BarTextMode.Single)
+                .SetBackgroundColor(EmptyColor["background"])
+                .SetText(BarTextPosition.CenterMiddle, BarTextType.Custom,
+                    (_config.ASTShowStarTextBar
+                        ? (!_config.ASTEnableDecimalStarBar
+                            ? starDuration.ToString("N0")
+                            : System.Math.Abs(starDuration).ToString("N1"))
+                        : ""));
+
+            if (starColorSelector == _config.ASTStarGiantColor.Map && _config.ASTShowStarGlowBar)
+            {
+                bar.SetGlowColor(_config.ASTStarGlowColor.Map["base"]);
             }
 
+            ;
+
             var drawList = ImGui.GetWindowDrawList();
             bar.Build().Draw(drawList, PluginConfiguration);
         }
     }
+
+    [Serializable]
+    public class AstrologianHudConfig : PluginConfigObject
```

Comment:
Awesome!
Really minor, but you could remove the "AST" prefix from the properties since its now redundant with the name of the class it contains them.